File: csharp/examples/GetInvoice.cs
using System;

namespace Examples.examples
{
    // Example of GET /invoices/:invoice_id

    class GetInvoice : BaseRequest
    {
        private string _invoiceId;

        public GetInvoice()
        {
            dotenv.net.DotEnv.Load();
            _invoiceId = Environment.GetEnvironmentVariable("INVOICE_ID");
            Console.WriteLine($"GetInvoice: invoice_id={_invoiceId}");
        }

        public GetInvoice(string invoiceId)
        {
            _invoiceId = invoiceId;
        }

        public override string SendRequest()
        {
            string path = $"invoices/{_invoiceId}";
            return GetRequest(path);
        }
    }
}

File: php/examples/get_invoice.php
<?php
require_once 'base_request.php';

// Example of GET /invoices/:invoice_id

$debug = true;
$invoice_id   = $_ENV['INVOICE_ID'];
$request_data = '';
$request_params = [];
// TODO: how Guzzle sends array query params is received by Ruby as indifferent access Hash
// $request_params = ['names' => ['Test Company 2']];

send_request('GET', '/invoices/' . $invoice_id, $request_params, $request_data);
